server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -152,6 +157,52 @@
 public class CompactionCoordinator
     implements CompactionCoordinatorService.Iface, Runnable, MetricsProducer {
 
+  // Object that serves as a TopN view of the RunningCompactions, ordered by
+  // RunningCompaction start time. The first entry in this Set should be the
+  // oldest RunningCompaction.
+  private static class TimeOrderedRunningCompactionSet
+      extends ConcurrentSkipListSet<RunningCompaction> {
+
+    // TODO: Could make this configurable, but probably should not be large. These
+    // are shown on the monitor, can't imagine that someone is going to page through
+    // all of them.
+    private static final int UPPER_LIMIT = 250;
+
+    private static final long serialVersionUID = 1L;
+
+    private final AtomicInteger size = new AtomicInteger(0);
+
+    TimeOrderedRunningCompactionSet() {
+      super(new Comparator<RunningCompaction>() {
+        @Override
+        public int compare(RunningCompaction rc1, RunningCompaction rc2) {
+          return Long.compare(rc1.getStartTime(), rc2.getStartTime());
```

Comment:
May want a secondary comparison on the external compaction uuid for the case where timestamps are equal.